### ISSUE TITLE
Data Sufficiency Test

### ISSUE THREAD

**Issue by suicee** [maho3/ltu-gobig#39]
As we discussed in several meetings, one possible reason why we do not obtain strong constraining power from higher-order statistics is that the training data may be insufficient. To test this idea, I ran the following experiment using the Quijote galaxy summaries.

I keep the test set fixed. For the training set, I construct a sequence in which the number of training samples is gradually increased. For each training-set size, I train an NPE model to convergence and then evaluate the log probability on the test set. The resulting plot for summary = bar + zpk0 suggests that increasing the number of training samples may further improve the constraining power. Different colors correspond to different trials, where the training sequences are constructed using different random seeds.

<img width="576" height="432" alt="Image" src="https://github.com/user-attachments/assets/43fc6b15-b3fc-4a3d-b6e3-47ad9de07a20" />

As a comparison, I ran a similar experiment using an [emulator](https://github.com/DeaglanBartlett/symbolic_pofk) to generate a large number of matter power spectra, adding noise to them. I then repeat the same procedure: gradually increasing the training-set size and evaluating the test log probability. In this case, the log probability converges to a fixed value beyond a certain number of training samples and does not improve further. This converged value is essentially the mutual information between the summaries and the parameters.

<img width="562" height="432" alt="Image" src="https://github.com/user-attachments/assets/c6587841-8e2f-42da-9979-999db0e768f9" />

Combining these two experiments, this suggests that for the Quijote galaxy summaries we are likely still in a data-limited regime, and increasing the number of training samples could improve constraints.

Note that the test log-probability curve also depends on the choice of model and training setup. For example, a model that is better aligned with the true posterior may require fewer training samples (e.g., using a Gaussian model to fit an approximately Gaussian posterior). Therefore, improving the model configuration itself may also help alleviate this issue.

**Comment by suicee** [comment]
Results for nbar + pk0 using the newly generated 4,000 Quijote-like FastPM simulations (only ~3,300 actually used, with 5 HOD parameters per simulation). The convergence looks improved. We should rerun this either with all 4,000 simulations or once the final setup for training the NPEs is fixed.

<img width="576" height="432" alt="Image" src="https://github.com/user-attachments/assets/375b3f71-e597-44ba-aa4f-78efb4f30a16" />
